core/market/src/protocol/negotiation/error.rs

```diff
@@ -28,24 +29,43 @@ pub enum CounterProposalError {
     NoPrevious(ProposalId),
     #[error("Countering Proposal [{1}] remote error: {0}")]
     Remote(RemoteProposalError, ProposalId),
+    #[error("Remote error: {0}")]
+    RemoteInternal(#[from] RemoteProposalError),
+}
+
+#[derive(Error, Debug, Serialize, Deserialize)]
+
+pub enum ReasonError {
+    #[error("Reason {0} serialize error: {1}")]
+    Serialize(Reason, String),
+    #[error("Reason `{0}` deserialize error: {1}")]
+    Deserialize(String, String),
+}
+
+#[derive(Error, Debug, Serialize, Deserialize)]
+pub enum RejectProposalError {
+    #[error("Rejecting {0}.")]
+    Gsb(#[from] GsbProposalError),
+    #[error(transparent)]
+    Get(#[from] GetProposalError),
+    #[error(transparent)]
+    Reason(#[from] ReasonError),
+    #[error(transparent)]
+    ChangeState(#[from] ChangeProposalStateError),
+    #[error(transparent)]
+    Validation(#[from] ProposalValidationError),
 }
 
 #[derive(Error, Debug, Serialize, Deserialize)]
 pub enum RemoteProposalError {
-    #[error("Offer/Demand [{0}] already unsubscribed.")]
-    Unsubscribed(SubscriptionId),
-    #[error("Offer/Demand [{0}] expired.")]
-    Expired(SubscriptionId),
+    #[error(transparent)]
+    Validation(#[from] ProposalValidationError),
```

Comment:
You hide important errors behind `ProposalValidationError`. If someone wants to match on something, he would like to match exactly on `Unsubscribe`, `Expired` and so on.

Reply:
It is still possible. Look within the tests. Or am I missing sth?

Reply:
explained on videochat